convert2rhel/main.py

```diff
@@ -195,65 +209,49 @@ def show_eula():
-        loggerinst.task("Convert: Subscription Manager - Verify installation")
-        subscription.verify_rhsm_installed()
-        loggerinst.task("Convert: Install RHEL certificates for RHSM")
-        system_cert = cert.SystemCert()
-        system_cert.install()
-
-    # remove non-RHEL packages containing repofiles or affecting variables in the repofiles
-    # This needs to be before attempt to unregister the system because after unregistration can be lost
-    # access to repositories needed for backuping removed packages
-    loggerinst.task("Convert: Remove packages containing .repo files")
-    pkghandler.remove_repofile_pkgs()
-
-    if not toolopts.tool_opts.no_rhsm:
-        loggerinst.task("Convert: Subscription Manager - Subscribe system")
-        subscription.subscribe_system()
-        loggerinst.task("Convert: Get RHEL repository IDs")
-        rhel_repoids = repo.get_rhel_repoids()
-        loggerinst.task("Convert: Subscription Manager - Check required repositories")
-        subscription.check_needed_repos_availability(rhel_repoids)
-        loggerinst.task("Convert: Subscription Manager - Disable all repositories")
-        subscription.disable_repos()
-
-    # we need to enable repos after removing repofile pkgs, otherwise we don't get backups
-    # to restore from on a rollback
-    if not toolopts.tool_opts.no_rhsm:
-        loggerinst.task("Convert: Subscription Manager - Enable RHEL repositories")
-        subscription.enable_repos(rhel_repoids)
-
-    # perform final checks before the conversion
-    loggerinst.task("Convert: Final system checks before main conversion")
-    checks.perform_pre_ponr_checks()
+#
+# Gathering system information
+#
+
+
+def gather_system_info():
+    # gather system information
+    loggerinst.task("Prepare: Gather system information")
+    systeminfo.system_info.resolve_system_info()
+    systeminfo.system_info.print_system_information()
+    breadcrumbs.breadcrumbs.collect_early_data()
+
+
+def prepare_system():
+    loggerinst.task("Prepare: Clear YUM/DNF version locks")
+    pkghandler.clear_versionlock()
+
+    loggerinst.task("Prepare: Clean yum cache metadata")
+    pkgmanager.clean_yum_metadata()
+
+
+def post_ponr_changes():
+    loggerinst.info("Starting Conversion")
+    post_ponr_conversion()
+
+    loggerinst.task("Final: Show RPM files modified by the conversion")
+    systeminfo.system_info.modified_rpm_files_diff()
+
+    loggerinst.task("Final: Update GRUB2 configuration")
+    grub.update_grub_after_conversion()
+
+    loggerinst.task("Final: Remove temporary folder %s" % utils.TMP_DIR)
+    utils.remove_tmp_dir()
+
+    loggerinst.task("Final: Check kernel boot files")
+    checks.check_kernel_boot_files()
+
+    breadcrumbs.breadcrumbs.finish_collection(success=True)
+
+    loggerinst.task("Final: Update RHSM custom facts")
+    subscription.update_rhsm_custom_facts()
+
+    loggerinst.info("\nConversion successful!\n")
```

Comment:
Thinking about this message here, maybe we should move it to be after this function call in `main()`? I have the impression that this message is not fitting here as part of "execute something", but as more of an informative way, with that in mind, I tend to think that we should move this message (and the one at the beginning of the function) to be in the main workflow, instead of here. 

What do you think?

Reply:
I agree.  Moved this to `main()`